frontend/src/views/InvoiceView.vue

```diff
@@ -0,0 +1,236 @@
+import type { Order } from "@/global/types/entities";
+import { calculateItemSumPrice, formatDateToIsoString } from "@/helpers/utils";
+import { ROUTES } from "@/router";
+
+const route = useRoute();
+const toast = useToast();
+
+let itemCount = 1;
+let calendarCount = 1;
+
+let invoiceItemsArray = ref<InvoiceItemCreate[]>([]);
+
+let invoiceInfo = ref({
+  order_id: "",
+  status: PaymentStatus.initial,
+  description: "",
+  invoice_date: "",
+  payment_target: "",
+  items: [],
+});
+
+let orderInfo = ref<Order | undefined>();
+const invoiceType = Object.values(PaymentStatus);
+let invoiceDate = ref<Date>();
+let paymentTarget = ref<Date>();
+let serviceDates = ref<{ id: number; date?: Date }[]>([{ id: 0 }]);
+
+const items = [
+  {
+    label: "Hinweis hinzugefügen",
+    command: () => {
+      onItemCreate(ArticleKind.heading);
+      toast.add({ severity: "success", detail: "Hinweis hinzugefügt", life: 3000 });
+    },
+  },
+  {
+    label: "Position hinzugefügen",
+    command: () => {
+      onItemCreate(ArticleKind.item),
+        toast.add({ severity: "success", detail: "Position hinzugefügt", life: 3000 });
+    },
+  },
+];
+
+function onServiceDateCreate() {
+  serviceDates.value.push({ id: calendarCount++ });
+}
+
+function onItemDelete(id: number) {
+  invoiceItemsArray.value = invoiceItemsArray.value.filter((element) => element.id !== id);
+}
+
+function onItemUpdate(item: InvoiceItemCreate) {
+  invoiceItemsArray.value = invoiceItemsArray.value.map((element) => {
+    if (element.id === item.id) {
+      return item;
+    } else {
+      return element;
+    }
+  });
+}
+
+function onItemCreate(kind: ArticleKind) {
+  invoiceItemsArray.value.push({ id: itemCount++, kind, title: "", description: "" });
+}
+function onServiceDateDelete(id: number) {
+  serviceDates.value = serviceDates.value.filter((element) => element.id !== id);
+}
+const allItemsSum = computed(() => {
+  return calculateItemSumPrice(invoiceItemsArray.value, invoiceInfo.value.invoice_date);
+});
+
+watch(invoiceDate, () => {
+  if (invoiceDate.value) {
+    invoiceInfo.value.invoice_date = formatDateToIsoString(invoiceDate.value);
+  } else {
+    invoiceInfo.value.invoice_date = "";
+  }
+});
+
+watch(paymentTarget, () => {
+  if (paymentTarget.value) {
+    invoiceInfo.value.payment_target = formatDateToIsoString(paymentTarget.value);
+  } else {
+    invoiceInfo.value.payment_target = "";
+  }
+});
+
+onMounted(async () => {
+  orderInfo.value = await getOrder(route.params.order_id as string);
+  invoiceInfo.value.order_id = orderInfo.value.id;
+});
+</script>
+<template>
+  <div class="flex flex-row justify-between">
+    <router-link :to="`${ROUTES.ORDER.path}/${route.params.order_id}/edit`">
+      <Button icon="pi pi-arrow-left" size="small" severity="secondary" text raised />
+    </router-link>
+    <div class="flex gap-x-2">
+      <Button label="Speichern" text raised />
+      <Button label="Löschen" severity="danger" text raised />
```

Comment:
Shouldn't these buttons have associated handlers?

Reply:
I will add it later when I do backend integration